packages/coordinator/src/client-api.ts

```diff
@@ -0,0 +1,141 @@
+import { Bytes32 } from 'soltypes'
+import { Block } from '@zkopru/core'
+import { CoordinatorContext } from './context'
+import EthMethods from './eth-rpc-methods'
+
+export class ClientApi {
+  context: CoordinatorContext
+
+  rpcMethods: { [key: string]: Function }
+
+  constructor(context: CoordinatorContext) {
+    this.context = context
+    /* eslint-disable @typescript-eslint/camelcase */
+    this.rpcMethods = {
```

Comment:
I think it'll be good to type the rpc methods just like
```
interface ZkopruRPCMethods {
    l1_address: ~,
    l2_*****: ~,
}
```

Reply:
I don't think it makes sense to type the RPC methods like this. The methods aren't meant to be called directly, only by passing the string `method` in an RPC call.

It might make sense to type the method names like this:

```ts
const APIMethods = {
  getAddress: 'l1_getAddress',
  getBlockByNumber: 'l2_getBlockByNumber',
  getBlockByHash: 'l2_getBlockByHash',
  ...
}
```

I'll keep something like this in mind when building the client lib.